tests/subsys/shell/shell/src/main.c

```diff
@@ -373,6 +373,38 @@ static void test_raw_arg(void)
 	shell_set_root_cmd(NULL);
 }
 
+#define CMD_MAND_OPT_RAW test_cmd_mand_opt_raw
+
+static int cmd_mand_opt_raw(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 2 || argc > 3) {
+		return -1;
+	}
+
+	if (strcmp(argv[0], STRINGIFY(CMD_MAND_OPT_RAW))) {
+		return -1;
+	}
+
+	if (argc >= 2 && strcmp(argv[1], "mandatory")) {
+		return -1;
+	}
+
+	if (argc == 3 && strcmp(argv[2], RAW_ARG)) {
+		return -1;
+	}
+
+	return 0;
+}
+
+SHELL_CMD_ARG_REGISTER(CMD_MAND_OPT_RAW, NULL, NULL, cmd_mand_opt_raw, 2, SHELL_OPT_ARG_RAW);
+
+static void test_mand_opt_raw(void)
```

Comment:
there is already a test with `SHELL_OPT_ARG_RAW` but with 1 mandatory argument (test_raw_arg). Can you merge those tests or rename somehow so they are similar (e.g. test_mand_1_opt_raw and test_mand_2_opt_raw).

Reply:
I ended up keeping the two tests separate and named them following your suggestion: `test_mand_1_opt_raw` and `test_mand_2_opt_raw`